argocd/resource_argocd_project.go

```diff
@@ -164,7 +164,7 @@ func resourceArgoCDProjectUpdate(ctx context.Context, d *schema.ResourceData, me
 		return errorToDiagnostics(fmt.Sprintf("failed to expand project %s", d.Id()), err)
 	}
```

Comment:
```sugggestion
	}

	if !si.IsFeatureSupported(ctx, features.ProjectDestinationServiceAccounts) {
		_, destinationServiceAccountsOk := d.GetOk("spec.0.destination_service_account")
		if destinationServiceAccountsOk {
			return featureNotSupported(features.ProjectDestinationServiceAccounts)
		}
	}
```

Please check if the feature is enabled during creation/update if the configuration is used.